src/t8_forest/t8_forest_adapt.cxx

```diff
@@ -245,26 +252,49 @@ t8_forest_adapt (t8_forest_t forest)
      * array which could be coarsened recursively. */
     el_coarsen = 0;
     /* TODO: this will generate problems with pyramidal elements */
+
+    /*Use num_siblings*/
     num_children =
       tscheme->t8_element_num_children (t8_element_array_index_locidx
                                         (telements_from, 0));
+    max_num_siblings = tscheme->t8_element_max_num_siblings(t8_element_array_index_locidx
+                                                   (telements_from, 0));
```

Comment:
We should use num_siblings here instead and regrow the array if necessary using realloc.

Why? The maximum number of siblings may be significantly larger than the actual number of needed elements for refinement.
When we introduce subelements for example max num children for quads could be 9, while most quads only generat 4 children.

Reply:
Has already been changed to num_siblings